### DISCUSSION TITLE
cannot render form-entries

### DISCUSSION THREAD

**Discussion by bastianborndoerfer** [neuefische/questions#705] [Web]
Hi everyone, I need help! I want to transfer the data entered in the form from the watchlist page to the portfolio page and render the entries there. I have tried it with .map and also with useRouter, but there is always an error somewhere that I cannot find. Is it because I am not handling the form through an extra intermediate page, or in the onChange function? It could be that it's my cold, but I can't seem to get any further.

[Here](https://github.com/bastianborndoerfer/capstone-project/tree/feat/add-assets) you can find my current code.  the branch is called: "feat/add-asset":
I hope someone can help :-) Maybe it's just a small mistake. If not, I'll try to rebuild everything and get to the goal with an extra form page and the corresponding components.

**Comment by F-Kirchhoff** [comment]
You need to "lift up" your state formData to the global "_app". This special component can pass props to `every single page`. Each page receives the props you pass the the `Component` inside the `_app`. 

Then you can pass the state and `handleFormChange` function from the `_app` into your `Component` and receive the props in the `watchlist` as well as the `portfolio` page: 

```js
// _app.js

// move your state and all related handler function here
return <Component {...pageProps} formData={formData} onFormChange={handleFormChange} />

//watchlist.js
export default function Page({formData, onFormChange}) {
 // use the two props in your form
}

//portfolio.js
export default function Page({formData}) {
 // use the form data here as well
}
```
Maybe you need to save the entered data from the form in a new state first. Then you need to define that state in the _app.js as well and pass down the state and the handler functions in the same way. 

In general, every time you need the same information on multiple pages, you need to move the related state into the _app.js

If you need another example, please have a look at our `global state` session.
